### PR TITLE
feat: emit pod events on drain

### PR DESCRIPTION
**Pull request by dthorsen**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
LGTM - Paired and tested these events with @dthorsen .

File: pkg/node/node.go [hunk 1]
```diff
@@ -58,6 +59,13 @@ const (
 	maxTaintValueLength = 63
 )
 
+const (
```
Comment (trutx) [inline pkg/node/node.go:62]: For consistency, should these also live in `pkg/observability/k8s_events.go`?

File: pkg/node/node.go [hunk 2]
```diff
@@ -669,3 +696,7 @@ func getUptimeFunc(uptimeFile string) uptime.UptimeFuncType {
 	}
 	return uptime.Uptime
 }
+
+type recorderInterface interface {
```
Comment (trutx) [inline pkg/node/node.go:700]: Should this type live in `pkg/observability` too?

File: pkg/node/node.go [hunk 3]
```diff
@@ -105,6 +113,25 @@ func (n Node) CordonAndDrain(nodeName string) error {
 	if err != nil {
 		return err
 	}
+	// Emit events for all pods that will be evicted
+	if recorder != nil {
```
Comment (trutx) [inline pkg/node/node.go:117]: I know if `nthConfig.EmitKubernetesEvents == false` then `recorder == nil` , but I think we should pass the entire `nthConfig` in. One reason is for readability (I suspect upstream folks will want that) and another reason is we could also stamp `nthConfig.KubernetesEventsExtraAnnotations` into the Pod events.

Comment (trutx) [reply pkg/node/node.go:117]: Or maybe just these two params would suffice, you get the point.

**Comment by trutx** [comment]
If you plan to PR this upstream I'd suggest you add an extra test here https://github.com/newrelic-forks/aws-node-termination-handler/blob/main/test/e2e/spot-interruption-test-events-on

**Pull request merged**